app.js

```diff
@@ -0,0 +1,129 @@
+// Application
+const app = (function () {
+  let tasks = [];
+
+  // Cache DOM
+  const form = document.querySelector('form');
+  const input = document.getElementById('text-input');
+  const list = document.getElementById('todo-list');
+  const sortByNameButton = document.getElementById('sort-by-name');
+  const sortByDateButton = document.getElementById('sort-by-date');
+
+  // Initialize
+  if (localStorage.getItem('tasks')) {
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+  }
+
+  function render () {
+    let data = {
```

Comment:
Why do you put tasks and html into `data` object? You're inside app module. You have access to tasks. For html just create variable `html`.